docs/language/contracts.rst

```diff
@@ -18,18 +18,24 @@ like so:
       }
   }
 
-A constructor does not have a name and may have any number of arguments. If a constructor has arguments,
-then when the contract is deployed then those arguments must be supplied.
+A constructor can have any number of arguments.
+If a constructor has arguments, they must be supplied when the contract is deployed.
 
-If a contract is expected to hold receive value on instantiation, the constructor should be declared ``payable``.
+If a contract is expected to receive value on instantiation, the constructor should be declared ``payable``.
 
 .. note::
+    On Substrate, constructors have a name. Solang allows to name constructors in the substrate target:
```

Comment:
```suggestion
    On Substrate, constructors have a name. Solang allows naming constructors in the substrate target:
```
See: https://english.stackexchange.com/questions/60271/grammatical-complements-for-allow/60285#60285